File: RingRing-Java/src/main/java/be/ringring/message/model/response/MessageResponse.java
package be.ringring.message.model.response;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * POJO with message response data
**/
public class MessageResponse extends AbstractResponse {

	
	//--- PROPERTIES -------------------------------------//
	
	@JsonProperty("MessageCount")
	private Integer messageCount;
	
	@JsonProperty("Reference")
	private String reference;
	
	@JsonProperty("Messages")
	private List<Message> messages;

	
	//--- GETTERS AND SETTERS ----------------------------//
	
	public Integer getMessageCount() {
		return messageCount;
	}

	public void setMessageCount(Integer messageCount) {
		this.messageCount = messageCount;
	}

	public String getReference() {
		return reference;
	}

	public void setReference(String reference) {
		this.reference = reference;
	}

	public List<Message> getMessages() {
		return messages;
	}

	public void setMessages(List<Message> messages) {
		this.messages = messages;
	}
	
	
}

File: RingRing-CSharp/DataObject/MessageResponse.cs
using System.Collections.Generic;

namespace RingRing
{
    /// <summary>
    /// Response object
    /// </summary>
    
    public class MessageResponse
    {
        public MessageResponse()
        {
        }

        /// <summary>
        /// Count of message
        /// </summary>
        public int MessageCount { get; set; }

        /// <summary>
        /// Your Reference
        /// </summary>
        public string Reference { get; set; }

        public List<Message> Messages { get; set; }

        public int ResultCode { get; set; }

        public string ResultDescription { get; set; }
    }

   

    public class Messages : List<Message>
    {
            
    }
}